src/lib/util/opresolv.h

```diff
@@ -102,61 +136,98 @@ class option_resolution
 		INTERNAL
 	};
 
+	template<typename T>
 	struct range
 	{
-		int min, max;
+		range() { min = -1; max = -1; }
+		T min, max;
 	};
 
-	option_resolution(const option_guide *guide, const char *specification);
-	~option_resolution();
-
-	// processing options with option_resolution objects
-	error add_param(const char *param, const std::string &value);
-	error finish();
```

Comment:
`finish` previously allowed a runtime check that the structure was valid.  You're now relying on asserts which are only compiled in debug builds.  This strikes me as rather dangerous.  Can you explain your thinking?